### ISSUE TITLE
Model Variations  

### ISSUE THREAD

**Issue by affect1210** [affect1210/Deep-Learning#5]
[Convolutional Neural Networks for Sentence Classification.pdf](https://github.com/affect1210/Deep-Learning/files/2801193/Convolutional.Neural.Networks.for.Sentence.Classification.pdf)
**Model Variations**
We experiment with several variants of the model.
• **CNN-rand:** Our baseline model where all words are randomly initialized and then mod-ified during training.
• **CNN-static:** A model with pre-trained vectors from _word2vec_. All words - including the unknown ones that are randomly initialized—are kept static and only the other parameters of the model are learned.
• **CNN-non-static:** Same as above but the pre-trained vectors are fine-tuned for each task.
• **CNN-multichannel:** A model with two sets of word vectors. Each set of vectors is treated as a ‘channel’ and each filter is applieto both channels, but gradients are back-propagated only through one of the chan-
nels. Hence the model is able to fine-tune one set of vectors while keeping the other static. Both channels are initialized with word2vec.
